controllers/servingruntime_controller.go

```diff
@@ -94,21 +94,47 @@ type runtimeInfo struct {
 func (r *ServingRuntimeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	log := r.Log.WithValues("servingruntime", req.NamespacedName)
 	log.V(1).Info("ServingRuntime reconciler called")
-
+	log.Info("=== ChinDebug sr reconcile 0 ===", "req.Namespace", req.Namespace)
 	// Make sure the namespace has serving enabled
 	mmEnabled, err := modelMeshEnabled2(ctx, req.Namespace, r.ControllerNamespace, r.Client, r.HasNamespaceAccess)
 	if err != nil {
 		return RequeueResult, err
 	}
 	runtimes := &kserveapi.ServingRuntimeList{}
+	csrs := &kserveapi.ClusterServingRuntimeList{}
+	if r.HasNamespaceAccess {
+		if err = r.Client.List(ctx, csrs); err != nil {
+			return RequeueResult, err
+		}
+	}
 	if mmEnabled {
 		if err = r.Client.List(ctx, runtimes, client.InNamespace(req.Namespace)); err != nil {
 			return RequeueResult, err
 		}
 	}
+	log.Info("=== ChinDebug sr reconcile 1 ===", "runtimes", runtimes)
+	log.Info("=== ChinDebug sr reconcile 1 ===", "len(runtimes.Items)", len(runtimes.Items))
+
+	// build the map for ServingRuntimeSpec
+	srSpecMap := make(map[string]kserveapi.ServingRuntimeSpec)
+	for i := range csrs.Items {
+		crt := &csrs.Items[i]
+		if crt.Spec.IsMultiModelRuntime() {
+			srSpecMap[crt.GetName()] = crt.Spec
+		}
+	}
+	// rt spec would override crt spec by design
+	for i := range runtimes.Items {
+		rt := &runtimes.Items[i]
+		if rt.Spec.IsMultiModelRuntime() {
+			srSpecMap[rt.GetName()] = rt.Spec
+		}
+	}
```

Comment:
srSpecs[rt.GetName()] = &runtimes.Items[i].Spec won't work because rt is not defined. It could become `srSpecs[runtimes.Items[i].GetName()] = &runtimes.Items[i].Spec` Not sure it is better that what it is now. I will leave it unchanged.